### DISCUSSION TITLE
Whole Code for X3D

### DISCUSSION THREAD

**Discussion by abhishek0696** [open-mmlab/mmaction2#1322] [Design]
HI, first of all, thanks for the comprehensive codebase for all these DL models. I wanted to know when will the whole code for X3D be available? I replicated the config file for X3D same as TPN, but there seems to be other discontinuity in the codebase for the same, and it is giving me errors given in the screenshot. It would be great to have the whole code for X3D!

![image](https://user-images.githubusercontent.com/63160543/145684768-61b77d66-89fb-4378-83a4-2582646a5462.png)

Waiting for a response. Please mail in abhishek.shivdeo98@gmail.com if I don't reply back.
Thanks a lot again!

**Comment by dreamerlin** [comment]
It seems you miss this line in your config file. https://github.com/open-mmlab/mmaction2/blob/master/configs/_base_/default_runtime.py#L10

**Reply by abhishek0696** [reply]
Hello @dreamerlin, thanks for the reply, even I think so, the thing is that the config file in __base__/models/x3d.py seems to be incomplete. I have attached the SS for the models base .py file for x3d and tpn for comparison. x3d
 is incomplete whereas tpn.py for base/models is working. Everything else is the same in both the models and tpn is working whereas x3d is not.
So is it possible to get a complete base/models/x3d.py config file for x3d?! It would make the code run! WOULD LOVE TO GET THE WHOLE CODE (run files as well as configs files) FOR X3D!

![Screenshot from 2021-12-17 23-10-46](https://user-images.githubusercontent.com/63160543/146628467-a61fa797-a126-46a2-ad07-32de3ad0b2cd.png)
![Screenshot from 2021-12-17 23-05-49](https://user-images.githubusercontent.com/63160543/146628468-4cadfdb0-2b79-421c-b711-34680b1c7ada.png)
